FC-Chess/FC-Chess.cpp

```diff
@@ -107,28 +105,271 @@ void FCC::Chess::checkMovesHorizontal()
 {
     for (uint8_t i = selX; i <= 7; i++)
     {
-        if (checkIsEnemyOnField(&i, &selY)) break;
+        if (checkIsFriendlyOnField(&i, &selY)) break;
         posPossible[i][selY] = true;
+        if (checkIsEnemyOnField(&i, &selY)) break;
     }
     for (uint8_t i = selX; i >= 0; i--)
     {
-        if (checkIsEnemyOnField(&i, &selY)) break;
+        if (checkIsFriendlyOnField(&i, &selY)) break;
         posPossible[i][selY] = true;
+        if (checkIsEnemyOnField(&i, &selY)) break;
     }
 }
 
 void FCC::Chess::checkMovesVertical()
 {
     for (uint8_t i = selY; i <= 7; i++)
     {
-        if (checkIsEnemyOnField(&selX, &i)) break;
+        if (checkIsFriendlyOnField(&selX, &i)) break;
         posPossible[selX][i] = true;
+        if (checkIsEnemyOnField(&selX, &i)) break;
     }
     for (uint8_t i = selY; i >= 0; i--)
     {
-        if (checkIsEnemyOnField(&selX, &i)) break;
+        if (checkIsFriendlyOnField(&selX, &i)) break;
         posPossible[selX][i] = true;
+        if (checkIsEnemyOnField(&selX, &i)) break;
+    }
+}
+
+void FCC::Chess::checkMovesDiagonal()
```

Comment:
Consider doing:
```cpp
// checking possible moves
for(uint8_t i = selX; i < 8; i++) {
  for (uint8_t j = selY; j < 8; j++) {
    if (checkIsFriendlyOnField(&i, &j)) break;
    posPossible[i][j] = true;
    if (checkIsEnemyOnField(&i, &j)) break;
  }
}
```

Reply:
Had that thought as well, but that isn't the result I want. If I use two for loops, it'll go "up right CHECK right CHECK right  CHECK (...) Up right CHECK (...)". So it's checking every line before going one up.
But since I wanna check the fields in a scheme of "up right CHECK up right CHECK", so every position on the upper-right, I need to code it this way.

Up-left etc. the same

Reply:
Then do
```
// checking possible moves
for(uint8_t i = selX; i < 8; i++) {
    if (checkIsFriendlyOnField(&i, &i)) break;
    posPossible[i][j] = true;
    if (checkIsEnemyOnField(&i, &i)) break;
}
```

Basically use the i twice. When you need to opposite, just do
```
// checking possible moves
for(uint8_t i = selX; i < 8; i++) {
    if (checkIsFriendlyOnField(&i, SelX - i)) break;
    posPossible[i][j] = true;
    if (checkIsEnemyOnField( i, SelX - ii)) break;
}
```

Reply:
Oh, missed that. Imma implement it.

Reply:
Since I removed the pointers, I was able to apply this idea in a similar way. See upcoming pushes.